### ISSUE TITLE
[FR] Particle modal: save / apply behavior

### ISSUE THREAD

**Issue by thexmanxyz** [gantry/gantry5#2643]
Hey there, I noticed something in the usage workflow while using Gantry which should be improved to make the workflow more convenient.

**Example**
If I change a Particle and save it to review the changes. I only have these two options:

![grafik](https://user-images.githubusercontent.com/17965908/81376447-dd2f1080-9103-11ea-8bd1-bd2334ba024c.png)

**Both are closing the modal!**

**Comment by dmleeman** [comment]
Agreed, the current behaviour is not intuitive, in fact, there is no point in having two buttons as they both effectively behave the same! It is a minor but persistent annoyance. It certainly moves up to being distractingly annoying when making minor changes in more heavily feature packed particles.

**Comment by blizam** [comment]
They definitely do different things. Apply saves but doesn't publish and allows for the undo/redo buttons to do there thing. Save publishes.

Apply saves your settings within the Particle without writing the values to YAML. Using Apply only as you make changes allows you to use the Undo button. Apply is faster when trying to update a lot of particles as it's not taking the extra steps to actually save/publish the changes. The other thing Apply does, when you're inside of a Particle list item (like slides/items for the Owl particle or Content Items in the Content Cubes particle), it saves the individual settings of that item and closes it allow you to return to the particles main settings... If Apply left the item settings still visible you'd now have to use 2 clicks to save and close the settings; note also that there is no close button; only Apply or Cancel and Cancel obviously won't save your settings.

With Apply and Save, when I'm in an inner-item's settings, with a single click I can save and publish my changes to that item versus saving/applying the item settings, then saving the main particle.


I do sort of see what you're saying with Apply closing the modal for the Particle's main settings/initial modal. However, because there's no secondary close button (only Cancel) if Apply didn't close the modal I would think that would be equally confusing. Generally speaking main particle settings don't have a ton of options either so I guess it never bothered me to simply make all the changes I want to make before clicking apply; I'm not making so many changes that I have save-anxiety in-between changes.

**Comment by dmleeman** [comment]
@blizam You're absolutely right of course, I had my front end hat on during my last reply. 

And was working with some commercial Particles that offer a lot of options in Joomla.

Reading your post, the behaviour makes perfect sense! A gap in my thinking now filled

**Comment by thexmanxyz** [comment]
@dmleeman To leave the modal always open is definitively not how it should work. I'm aware that there are different usage scenarios. E.g.:

a.) I'm making 1-2 to changes in multiple e.g. atoms, particles and thats it
b.) I'm checking how a particle works and need to change different settings in the modal to verify them (which makes it necessary to leave that modal open).

b.) is quite frequent for complexer particles and there are a lot of them and a.) happens when there is a small set of settings. 

@blizam I'm on point with you about:

> The other thing Apply does, when you're inside of a Particle list item (like slides/items for the Owl particle or Content Items in the Content Cubes particle), it saves the individual settings of that item and closes it allowing you to return to the particle's main settings... If Apply left the item settings still visible you'd now have to use 2 clicks to save and close the settings; note also that there is no close button; only Apply or Cancel and Cancel obviously won't save your settings.

I initially pointed just on the "main modal". If opening and closing a collection list e.g. in Owl Carousel or Content Cubes this is a different thing. 

> With Apply and Save, when I'm in an inner-item's settings, with a single click I can save and publish my changes to that item versus saving/applying the item settings, then saving the main particle.

I'm not sure what you mean. Sub modals look like this:

![grafik](https://user-images.githubusercontent.com/17965908/82294289-2f96e980-99ae-11ea-95dc-bdc4e716077c.png)

You can't publish on sub elements. But that might be also considerable when modifying the current state or was that already what you meant?

TBH I'm also not very happy with leaving the modal open on "Apply and Save". It would improve the current state but it is not very intuitive. Three buttons is already enough. So we are limited on how to reflect the behaviour. 

But what would be a perfect solution to improve the current state? Allow users to globally change the behaviour (as kind of user setting)?

**Comment by thexmanxyz** [comment]
@blizam @dmleeman I updated my request with my new suggestion for the change.
